File: 338-counting-bits/338-counting-bits.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        dp = [0 for i in range(n+1)]
        
        for i in range(len(dp)):
            if i <= 1:
                dp[i] = i
            elif i % 2 == 0:
                dp[i] = dp[i//2]
            else:
                dp[i] = dp[i//2] + 1
        
        return dp

File: 338-counting-bits/338-counting-bits.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int>dp(n + 1);
        for (int i = 0; i <= n; i++) {
            if (i < 2)
                dp[i] = i;
            else
                dp[i] = dp[i/2] + (i & 1 ? 1 : 0);
        }
        return dp;
    }
};